base/complex.jl

```diff
@@ -424,7 +424,8 @@ function log2(z::Complex)
     a/log(oftype(real(a),2))
 end
 
-function exp(z::Complex)
+exp(z::Complex) = exp(float(z))
```

Comment:
I think I missunderstand you but the test above the one I added looks pretty similar to how I did it?